From d384e3236b6399562c301eb25269f7bc9ae066e9 Mon Sep 17 00:00:00 2001
From: fundamental <mark.d.mccurry@gmail.com>
Date: Wed, 3 Sep 2014 12:51:08 -0400
Subject: [PATCH 0235/1936] Nio: Fix Segfault In JackEngine

Adds Null checks in cleanup of jack engine code on close.
This fixes a possible segfault under -I jack -O oss with whatever is
currently installed on freebsd.
This patch is by Hans Petter Selasky
---
 src/Nio/JackEngine.cpp | 10 ++++++----
 1 file changed, 6 insertions(+), 4 deletions(-)

diff --git a/src/Nio/JackEngine.cpp b/src/Nio/JackEngine.cpp
index b0da311b..52fec4ea 100644
--- a/src/Nio/JackEngine.cpp
+++ b/src/Nio/JackEngine.cpp
@@ -238,16 +238,18 @@ void JackEngine::stopAudio()
     for(int i = 0; i < 2; ++i) {
         jack_port_t *port = audio.ports[i];
         audio.ports[i] = NULL;
-        if(NULL != port)
+        if(jackClient != NULL && NULL != port)
             jack_port_unregister(jackClient, port);
     }
     midi.jack_sync = false;
     if(osc.oscport) {
-        jack_port_unregister(jackClient, osc.oscport);
+       if (jackClient != NULL) {
+               jack_port_unregister(jackClient, osc.oscport);
 #ifdef JACK_HAS_METADATA_API
-        jack_uuid_t uuid = jack_port_uuid(osc.oscport);
-        jack_remove_property(jackClient, uuid, "http://jackaudio.org/metadata/event-types");
+               jack_uuid_t uuid = jack_port_uuid(osc.oscport);
+               jack_remove_property(jackClient, uuid, "http://jackaudio.org/metadata/event-types");
 #endif // JACK_HAS_METADATA_API
+       }
     }
     if(!getMidiEn())
         disconnectJack();
-- 
2.47.0

